pkg/controller/controller.go

```diff
@@ -102,8 +115,47 @@ func NewController(clientset kubernetes.Interface, ssclientset ssclientset.Inter
 		},
 	})
 
+	sInformer := sinformer.Core().V1().Secrets().Informer()
+	sInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: func(obj interface{}) {
+			skey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err != nil {
+				log.Printf("failed to fetch Secret key: %v", err)
+				return
+			}
+
+
+			if (!findOwnerReferenceSS(obj)) {
+				if (!isAnnotatedToBeManaged(obj.(*corev1.Secret))) {
+					return
+				}
+			}
+
+			ns, name, err := cache.SplitMetaNamespaceKey(skey)
+			if err != nil {
+				log.Printf("failed to split metadatada namespace: %v", err)
+				return
+			}
+
+			ssecret, err := ssclientset.BitnamiV1alpha1().SealedSecrets(ns).Get(context.Background(), name, metav1.GetOptions{})
+			if err != nil {
+				log.Printf("failed to find the Sealed Secret associated: %v", err)
+				return
+			}
+
+			sskey, err := cache.MetaNamespaceKeyFunc(ssecret)
+			if err != nil {
+				log.Printf("failed to fetch Sealed Secret key: %v", err)
+				return
+			}
```

Comment:
I am not sure about the second case but the first one seems like not found is perfectly normal, so I would do this instead:

```suggestion
			if err != nil {
			        if !errors.Is(err, NotFound) { // Only log if the error is unexpected at this point
				  log.Printf("failed to find the Sealed Secret associated: %v", err)
				}
				return
			}

			sskey, err := cache.MetaNamespaceKeyFunc(ssecret)
			if err != nil {
				log.Printf("failed to fetch Sealed Secret key: %v", err)
				return
			}
```

Reply:
With the isAnnotatedToBeManaged and the findOwnerReferenceSS, you are doing the checks previous to this code block. At this point, your secret must have always a sealed secret, if not you are having something not aligned in your cluster.